### PR TITLE
wire: improve color output to term by using fatih/color

### LINKED ISSUE #183: show: do not always print ANSI escape sequences
The `wire show` command always prints ANSI escape sequences to make the output coloured.
This is not appropriate when the terminal is not ANSI (for example, I usually use `TERM=dumb`) or when the command is being piped into another command.

### PR DESCRIPTION
**Pull request by vangent**
Fixes #183.

This addresses a TODO, and also ensures that the coloring works on Windows, and is disabled when appropriate (e.g., `TERM=dumb`, `wire show | less`).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,10 @@
 module github.com/google/wire
 
 require (
+	github.com/fatih/color v1.7.0
```
Comment (eliben) [inline go.mod]: Fly-by: this is adding to the dependency load of wire, which is being imported by clients. Maybe we should consider separate modules for the wire *library* and wire *command*  - the library should have very minimal dependencies (none?)

Comment (vangent) [reply go.mod]: @zombiezen 	WDYT? We could split `wire` into two modules....

Comment (zombiezen) [reply go.mod]: -1 here. While separating into separate modules works well for the Go CDK libraries, we actually stand to gain something here by keeping the code generator binary in the same module. Once modules is on by default, we can more easily ensure that the Wire code generator matches the runtime support library that is in use (see #160).

**Comment by rogpeppe** [comment]
FWIW personally I'd rather remove color support than incur a significant external dependency like this.

I'd like to be able to use wire code in low level packages with strict dependency requirements, but pulling in `github.com/fatih/color` also pulls in `github.com/mattn/go-colorable`, `github.com/mattn/go-isatty` and `golang.org/x/sys`.

Are the colors really worth it? Perhaps we can make the layout clearer without using color?

**Comment by vangent** [comment]
I filed https://github.com/google/wire/issues/189 to discuss dropping colors.

**Pull request merged**